tests/test_database_manager.py

```diff
@@ -0,0 +1,146 @@
+"""
+test_database_manager.py - tests for the DatabaseManager package.
+
+A database configuration and connection is required for these tests.
+No data will be written to or deleted from the database.
+
+Copyright (c) 2018 The Fuel Rats Mischief,
+All rights reserved.
+
+Licensed under the BSD 3-Clause License.
+
+See LICENSE
+"""
+import pytest
+import psycopg2
+from database import DatabaseManager
+from psycopg2 import extensions, pool, sql
+
+pytestmark = pytest.mark.database_manager
+
+
+@pytest.fixture
+def test_dbm() -> DatabaseManager:
+    """
+    Test fixture for Database Manager.
+
+    A DATABASE CONFIGURATION AND CONNECTION IS REQUIRED FOR THESE TESTS.
+    """
+    database = DatabaseManager()
+    return database
+
+
+@pytest.fixture()
+def test_dbm_pool(test_dbm) -> psycopg2.pool.SimpleConnectionPool:
+    """
+    Test fixture for Database Manager's connection pool.
+
+    A DATABASE CONFIGURATION AND CONNECTION IS REQUIRED FOR THESE TESTS.
+    """
+    return test_dbm._dbpool
+
+
+def test_dbm_singleton():
+    """
+    Verify that DBM is indeed a singleton and not spawning a new instance.
+    """
+    class Pirate(DatabaseManager):
+        pass
+
+    ship_crew = Pirate()
+    hating_sharks = Pirate()
+
+    assert ship_crew is hating_sharks
+
+
+def test_init_values(test_dbm):
+    """
+    Test and verify all values required for instantiation are not None.
+    """
+    assert test_dbm._initialized
+    assert test_dbm._dbhost
+    assert test_dbm._dbpass
+    assert test_dbm._dbname
+    assert test_dbm._dbport
+    assert test_dbm._dbuser
+
+
+def test_private_settings(test_dbm):
+    """
+    Assign values to
+    """
+
+
+def test_db_connection(test_dbm_pool):
```

Comment:
Is this testing our code, or testing `pyscopg`'s code?
Ditto for `test_db_connection_close`, `test_db_connection_autocommit`, and `test_db_connection_client_encoding`.

Reply:
I guess the scope has a little bit of overshadowing, but I felt it necessary to test the DBM's implementation and assert that these are actually valid, and being used.  When it comes to using live/production database I would like these to cause an error if they leave the library or become depreciated.

Reply:
Methinks i see where @Eearslya is coming from here, are we testing our code here or the psql library?
The behavior of dependencies alone should not be tested, unless we intend to run their test suite as well. It is beyond the scope of our testing